hexrd/ui/image_tab_widget.py

```diff
@@ -53,24 +56,13 @@ def setup_connections(self):
         HexrdConfig().detectors_changed.connect(self.reset_index)
 
     def clear(self):
-        # This removes all canvases except the first one,
-        # and it calls super().clear()
-
-        for canvas, cid in zip(self.image_canvases[1:],
-                               self.mpl_connections[1:]):
-            canvas.mpl_disconnect(cid)
-            canvas.deleteLater()
```

Comment:
Do you know why we were originally doing this? Is removing it going to have any side effects for other parts of the code?

Reply:
According to git blame this was to prevent a bug where signals were sometimes still connected to the canvases when the tabs were not being used ([here](https://github.com/HEXRD/hexrdgui/commit/043f1ef0d1006e090af02ec3b8eb9397d2407be2)). Without this change we face other issues (like we need to re-render the canvas with the masks but this re-creates the canvases and the signals for masking are broken). I think that if there is an issue with signals not being disconnected properly we should figure that out rather than just deleting and re-creating every time though.

Reply:
Yes I agree! Thanks for looking that up!